cmd/fs-v1-multipart.go

```diff
@@ -656,14 +661,21 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 	}
 
 	uploadIDPath := pathJoin(bucket, object, uploadID)
+	var removeObjectDir bool
 
 	// Hold the lock so that two parallel complete-multipart-uploads
 	// do not leave a stale uploads.json behind.
 	objectMPartPathLock := globalNSMutex.NewNSLock(minioMetaMultipartBucket, pathJoin(bucket, object))
 	if err = objectMPartPathLock.GetLock(globalOperationTimeout); err != nil {
 		return oi, err
 	}
-	defer objectMPartPathLock.Unlock()
+
+        defer func() {
+		objectMPartPathLock.Unlock()
+		if removeObjectDir == true {
```

Comment:
Suggestion:
The following is more idiomatic Go.
```go
if removeObjectDir {
}
```

Reply:
addressed